### PR TITLE
feat: scroll section detector (#233)

### PR DESCRIPTION
**Pull request by iamnamananand996**
will close this issue [https://github.com/instill-ai/instill.tech/issues/233](https://github.com/instill-ai/instill.tech/issues/233)

have exact same feature which is mentioned form **astro docs**

### CODE REVIEW

**Comment by iamnamananand996** [comment]
yes, it should close both the issue [https://github.com/instill-ai/instill.tech/issues/233](https://github.com/instill-ai/instill.tech/issues/233) and [https://github.com/instill-ai/instill.tech/issues/498](https://github.com/instill-ai/instill.tech/issues/498) as both have exactly same feature request

File: src/components/ui/TableOfContent.tsx [hunk 1]
```diff
@@ -13,28 +13,65 @@ export type TableOfContentProps = {
...
-  useEffect(() => {
-    if (!router.isReady) return;
+    const headingsObserver = new IntersectionObserver(
+      setCurrent,
+      observerOptions
+    );
+
+    // Observe all the headings in the main page content.
+    document
+      .querySelectorAll("article :is(h1,h2,h3)")
```
Comment (EiffelFly) [inline src/components/ui/TableOfContent.tsx:62]: @iamnamananand996 We didn't limit the usage of header, so the header may be h5 at maximum, we have to count this situation into consideration

```suggestion
      .querySelectorAll("article :is(h1,h2,h3,h4,h5)")
```

File: src/components/ui/TableOfContent.tsx [hunk 2]
```diff
@@ -13,28 +13,65 @@ export type TableOfContentProps = {
 
 export const TableOfContent = ({ headers }: TableOfContentProps) => {
   const router = useRouter();
+  const onThisPageID = "on-this-page-heading";
   const [currentHash, setCurrentHash] = useState<Nullable<string>>(null);
 
+  // useEffect(() => {
```
Comment (EiffelFly) [inline src/components/ui/TableOfContent.tsx:19]: @iamnamananand996 Let remove all these commented code

**Review by EiffelFly — COMMENTED** [review]
@iamnamananand996 Thanks for your contribution, I had added some suggestion~ Please review them

**Comment by iamnamananand996** [comment]
@EiffelFly, added the request changes.

**Review by EiffelFly — APPROVED** [review]
@iamnamananand996 well done!!! I am happy to merge this PR

**Pull request merged**